torch/_inductor/kernel/mm.py

```diff
@@ -164,9 +164,18 @@ def tuned_int_mm(mat1, mat2, *, layout=None):
     choices = (
         [aten__int_mm.bind((mat1, mat2), layout)] if use_aten_gemm_kernels() else []
     )
-    if m * n != 0 and use_triton_template(layout, enable_int32=True):
-        # TODO: Re-enable eager mode implementation once cuBLAS is fixed
+
+    # TODO: Re-enable eager mode implementation once cuBLAS is fixed
+    if m * n != 0 and (
+        use_cutlass_template(layout) or use_triton_template(layout, enable_int32=True)
+    ):
         choices = []
+
+    if m * n != 0 and use_cutlass_template(layout):
+        CUTLASSGemmTemplate.add_cutlass_gemm_choices(
```

Comment:
Where are the various configs defined? 

Below we use `int8_mm_configs` to see the various Triton kernels. Is there an equivalent for CUTLASS? If so, we likely need to update or unify that.

Reply:
For CUTLASS, these are specified within the CUTLASS itself, in the [generator](https://github.com/NVIDIA/cutlass/blob/main/python/cutlass_library/generator.py) code.

While the MMA PTX instructions eventually used to implement given operator should be the same for Triton and CUTLASS, there is number of possible differences: how are tiles transferred between global and shared memory, and then between shared memory and registers, then how is shared memory used so that single thread-block handles multiple tiles, etc.  For the same reason, CUTLASS generator suggests different tiles for different data-types combinations etc.  So I think there is no point in the unification.  (On the other side, while working on this and [other PR](https://github.com/pytorch/pytorch/pull/119986), I found some small omissions and inconsitencies in CUTLASS generator code, so I created [this PR](https://github.com/NVIDIA/cutlass/pull/1350) for CUTLASS to fix these - there may be more minor changes coming for CUTLASS generator like this).

Reply:
Ok, that makes sense, but we should have a sense for the number of configurations that are being tried.

Reply:
For Triton, I believe all of the configs listed are tried during auto-tuning.  CUTLASS [generator](https://github.com/NVIDIA/cutlass/blob/main/python/cutlass_library/generator.py) code typically enumerates 10-15 configurations for given operation, but a disadvantage of CUTLASS is that generated code is heavily templated C++, that takes long to compile.  For this reason, there is `cuda.cutlass_max_profiling_configs` parameter introduced into `torch._inductor.config`, making it possible for user to adjust the number of CUTLASS configurations tried during auto-tuning; the default setting for this parameter is that all CUTLASS configurations are tried too.  So this is a trade-off for CUTLASS, admittedly it's still somewhat unsatisfactory in the sense that if  `cuda.cutlass_max_profiling_configs` is set by user, in order to save compilation time, to less than number of configurations that CUTLASS generator lists for given operation, then there will be configurations listed by CUTLASS that won't be tried, and some among them could actually perform the best.  (For the record, I set this parameter to 10 for the benchmarking presented below.)

In general, I still think that down the road it would be nice to build a heuristic, probably the best approach would be some kind of machine learning, that could be trained up-front on various GPUs, and that would make it possible for a more educated guess on the good configuration(s) to try/use, for both eager and compiled mode.  It should not be hard to build, but would take time and resources to train, so it could be a nice idea for a side project.

Reply:
Agreed, that machine learning based approach ( a fast Decision Tree, maybe combined with a Linear Regression 
should suffice ) is also something that I suggested and was discussed a bit Pytorch internally.